### ISSUE TITLE
Downloaded references are GRCh38 but reads are mapped to h37

### ISSUE THREAD

**Issue by jeromekelleher** [ga4gh/server#635]
It seems that the ENA API that we're using only supports h38, so we need to find another way to programatically download a subset of the human references. Any ideas?

**Comment by macieksmuga** [comment]
@jeromekelleher could you expand on this issue description? 

Specifically, what are you using this ENA (I'm assuming European Nucleotide Archive) API for? Is it only for downloading test data, or is it used anywhere in the code? 

If the latter, this may be a problem, as most clinical variant data (including the data I'm including in the compliance suite) is still on GRCh37.

**Comment by diekhans** [comment]
is this a one-time download, or do you want to be able to programatically get subsequences?

ftp maybe the solution if its one time.

**Comment by jeromekelleher** [comment]
In the ``download_data.py`` script we download the specific pieces of the reference which we need to cover the subset of variants we want. Currently, this is the first 1000 variants on a handful of chromosomes. We then download the reference sequences from the [ENA REST APIs]( http://www.ebi.ac.uk/ena/browse/data-retrieval-rest). However, it turns out that ENA only has GRCh38 and we need GRCh37 for 1000 genomes data.

The current script is making life so much easier, it's just a pity we're using the wrong data. Downloading the entire reference just to get a few tiny specific fragments of sequence is absurd.

**Comment by jmarshall** [comment]
I see asking for e.g. [CM000670.**1**](http://www.ebi.ac.uk/ena/data/view/CM000670.1%26display=fasta&range=500000-501000) gives you a sequence entitled

```
>ENA|CM000670|CM000670.2 Homo sapiens chromosome 8, GRCh38 reference primary assembly. : Location:500000..501000
```
which seems a bit surprising — in fact [the documentation](http://www.ebi.ac.uk/ena/browse/data-retrieval-rest#sequence_version) suggests this is a bug.  Have you asked them about this?

Otherwise Ensembl has a REST API at http://grch37.rest.ensembl.org/ from which you can probably obtain the subsequences you need.

**Comment by jeromekelleher** [comment]
Thanks @jmarshall, I'll contact the ENA people and see what they say. They should really be returning an error if the version isn't present. I've taken a look through the Ensembl APIs, and it's not immediately obvious how to find what I'm looking for. Perhaps I need to dig a bit deeper.

**Comment by jmarshall** [comment]
Any news from ENA?

For Ensembl, either [sequence/id](http://grch37.rest.ensembl.org/documentation/info/sequence_id) with a range request (if it listens to that header) or [sequence/region](http://grch37.rest.ensembl.org/documentation/info/sequence_region) should do the trick I think.

**Comment by jeromekelleher** [comment]
No, had sort of phased out on this one. I'll get some definitive answers next week; thanks for the extra information @jmarshall.

**Comment by jeromekelleher** [comment]
This is now essentially the same issue as 719. Both can be closed if the issue is resolved.

**Comment by jsh2134** [comment]
The SolveBio API currently supports pulling this data free (albeit with different parameters) 

I'm one of the SolveBio devs (here at g4ahack) and would be happy to help incorporate it for #719. The sequence currently requires an API_KEY but we are planning on making it fully public

`$ curl -XGET "https://$SOLVEBIO_API_KEY@api.solvebio.com/v1/gws/sequence?genome_build=GRCh37&start=100000000&chromosome=12&stop=1000000000"`

```{"data": {"genomic_range": {"start": 100000000, "stop": 100000499, "chromosome": "12"}, "sequence": "ATATTCTCATTATTAACCCCTTGTCAGATGCATAGTTTACAAATGTTTTTTGGTATAACATAAGCATTTTTCCTATTTTTTAACAAAAATGTGTTCTTTATTCATTATGAAACATTTTACTTCCTCACAAAGAGTTGTTTCCTTTATAATTTCTAGTATTTTAATTACATGTATTAACTATAATTTATAACTTTTAGTAACTTAATTTCTAGTTAAAAGTAGCAAACTTTGTGAACTGTCTCATATTAGTACAAATGATTTCATTATTTCTAGAGACATGCATTTCCTCAATTTTTTACATTTATTAATAGATCCAAATATATTTAGCTTCTCTATATCTTATAAAAAGAAGACGCCAAAGTACATAAACTTATGAATAATCATATATGTTTTCATATTTTAACTTACTTAGAAATAACTCAGACAATTAATGAGTATCTCTATTACTTAATTTCACTATAAGGTTTCAAGTTACCAAAAAGAATTTTTAAACTATGACA"}}```

**Comment by reece** [comment]
I'm interested in this issue and #719.

I propose to use NCBI's eutils, which supports accession-based lookup and sequence slicing, and it's quick.

```time curl "http://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi?db=nucleotide&id=CM000670.1&rettype=fasta&seq_start=5000&seq_stop=5100"
>gb|CM000670.1|:5000-5100 Homo sapiens chromosome 8, GRCh37 primary reference assembly
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN
NNNNNNNNNNNNNNNNNNNNNNNNNNNNNNN

real	0m0.135s
user	0m0.012s
sys	0m0.004s
```

**Comment by jsh2134** [comment]
Yes! We will def remove registration and make the API spec public (although not for the purposes of this hackathon) - Def use NCBI if you're working on it today and once we make it easier on our end, we'll submit a PR.

Also we (SolveBio) are sitting by the Starbucks coffee if you wanna chat IRL

**Comment by reece** [comment]
Okay. I'm working on this one.

**Comment by reece** [comment]
I noticed that this issue was still assigned to me and open. My reading of the history is that this issue (#635) begat PR #754, which was rejected but spawned issue #768, which was resolved with PR #772.

So, I think this issue is stale. Closing.
